### ISSUE TITLE
Add Application Insight provisioning

### ISSUE THREAD

**Issue by veikkoeeva** [Azure/azure-libraries-for-net#354]
It appears the .NET Fluent Azure libraries do not support Application Insight provisioning while the Java SDK does. As per discussion at https://github.com/Azure/azure-sdk-for-net/issues/3384 this is just an unfortunate lapse. This issue is added for tracking the addition, and tagging @hovsepm due to suggestion in the previous thread.

There's prior discussion to this at https://github.com/Azure/azure-libraries-for-net/issues/210, this might unblock someone.

**Comment by hovsepm** [comment]
Just to be clear - Java Fluent SDK does not have App Insights. Sorry if I confused anyone. 

We will need to dedicate enough time to implement for both Java and C#.

**Comment by rajbos** [comment]
Finally got there after following all the links going through the Java 👎 . 

I really needed this functionality today, so a little disappointed that this wasn't available (more then a year after the first issue was opened!). Nevertheless, following along now ;-).

**Comment by tomkerkhove** [comment]
Looks like a duplicate of #27?

**Comment by rajbos** [comment]
I agree @tomkerkhove. This one should probably be closed (links to the Java libary will still be available here).

**Comment by hovsepm** [comment]
@rajbos if you are blocked I'll note that there is an autogenerated client for App Insights that you can give a try https://www.nuget.org/packages/Microsoft.Azure.Management.ApplicationInsights/0.1.1-preview

**Comment by rajbos** [comment]
Great! I’ll test with it 👍

**Comment by hovsepm** [comment]
Closed as by design.
